Sources/Orbit/Components/Icon.swift

```diff
@@ -55,6 +55,64 @@ public struct Icon: View {
                     .font(.system(size: size.value * Self.sfSymbolToOrbitSizeRatio * sizeCategory.ratio))
         }
     }
+
+    @available(iOS 14.0, *)
+    func textRepresentation(sizeCategory: ContentSizeCategory) -> SwiftUI.Text {
+        switch content {
+            case .symbol(let symbol, let color?):
+                return SwiftUI.Text(verbatim: symbol.value)
+                    .baselineOffset(-size.baselineOffset(sizeCategory: sizeCategory))
+                    .foregroundColor(color)
+                    .font(.orbitIcon(size: size.value, style: size.textStyle))
+            case .symbol(let symbol, nil):
+                return SwiftUI.Text(verbatim: symbol.value)
+                    .baselineOffset(-size.baselineOffset(sizeCategory: sizeCategory))
+                    // foregroundColor(nil) prevents further overrides
+                    .font(.orbitIcon(size: size.value, style: size.textStyle))
+            case .image(let image, _):
+                return SwiftUI.Text(image.resizable())
+            case .countryFlag(let countryCode):
+                return SwiftUI.Text(SwiftUI.Image(countryCode, bundle: .current).resizable())
+            case .sfSymbol(let systemName, let color?):
+                return SwiftUI.Text(Image(systemName: systemName)).foregroundColor(color)
+                    .font(.system(size: size.value * Self.sfSymbolToOrbitSizeRatio * sizeCategory.ratio))
+            case .sfSymbol(let systemName, nil):
+                return SwiftUI.Text(Image(systemName: systemName))
+                    .font(.system(size: size.value * Self.sfSymbolToOrbitSizeRatio * sizeCategory.ratio))
+        }
+    }
+
+    func textRepresentationFallback(sizeCategory: ContentSizeCategory) -> SwiftUI.Text? {
+        switch content {
+            case .symbol(let symbol, let color?):
+                return SwiftUI.Text(verbatim: symbol.value)
+                    .baselineOffset(-size.baselineOffset(sizeCategory: sizeCategory))
+                    .foregroundColor(color)
+                    .font(.orbitIcon(size: size.value, style: size.textStyle))
+            case .symbol(let symbol, nil):
+                return SwiftUI.Text(verbatim: symbol.value)
+                    .baselineOffset(-size.baselineOffset(sizeCategory: sizeCategory))
+                    // foregroundColor(nil) prevents further overrides
+                    .font(.orbitIcon(size: size.value, style: size.textStyle))
+            case .image, .countryFlag, .sfSymbol:
+                assertionFailure(".image, .countryFlag, .sfSymbol as Text are only available in iOS 14.0 or newer")
+                return nil
+        }
+    }
+}
+
+extension Icon: SwiftUITextRepresentable {
+    public func asText(configuration: ContentSizeCategory) -> SwiftUI.Text? {
+        if content.isEmpty == false {
```

Comment:
Since this is not a result builder, maybe let's use an early exit?

```swift
if content.isEmpty { return nil }
```

Reply:
replaced with `guard`